index.js

```diff
@@ -0,0 +1,68 @@
+const addon = require('bindings')('glue');
+
+const pkcs7_header = Buffer.from('-----BEGIN PKCS7-----');
+const pkcs7_footer = Buffer.from('-----END PKCS7-----');
+const nl = Buffer.from('\n');
+
+/**
+ * Verify a document given a public key, document and a PKCS#7 encoded
+ * signature.  All parameters should either be strings, Buffers or something
+ * which can be converted into a Buffer safely with a call to Buffer.from(),
+ * with an encoding parameter of 'utf-8'.
+ */
+module.exports = function verify(pubkey, document, pkcs7) {
+  if (typeof pubkey === 'undefined') {
+    throw new Error('pubkey must be provided');
+  }
+
+  if (typeof document === 'undefined') {
+    throw new Error('document must be provided');
+  }
+
+  if (typeof pkcs7 === 'undefined') {
+    throw new Error('pkcs7 signature must be provided');
+  }
+
+  if (!Buffer.isBuffer(pubkey)) {
+    pubkey = Buffer.from(pubkey, 'utf-8');
+  }
+
+  if (!Buffer.isBuffer(document)) {
+    document = Buffer.from(document, 'utf-8');
+  }
+
+  if (!Buffer.isBuffer(pkcs7)) {
+    pkcs7 = Buffer.from(pkcs7, 'utf-8');
+  }
+
+  hasHeader = false;
```

Comment:
nit: add a `let =` here?

Reply:
good catch.  I've actually added a 'use strict' directive now, thanks for the reminder :)